source/includes/fundamentals/code-snippets/Monitoring.java

```diff
@@ -0,0 +1,169 @@
+package com.mycompany.app;
+
+import com.mongodb.ConnectionString;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.ReadPreference;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.event.*;
+import org.bson.Document;
+
+import java.util.concurrent.TimeUnit;
+
+public class Monitoring {
+
+    private static final String COLLECTION = "compound-ops";
+    private static final String DATABASE = "test";
+    private static final ConnectionString URI = new ConnectionString(System.getenv("DRIVER_URL"));
+
+    public static void main(String[] args) throws InterruptedException {
+        Monitoring examples = new Monitoring();
+        System.out.println("\n---Command Event---\n");
+        examples.monitorCommandEvent();
+        System.out.println("\n---Cluster Event---\n");
+        examples.monitorClusterEvent();
+        System.out.println("\n---Connection Pool Event---\n");
+        examples.monitorConnectionPoolEvent();
+    }
+
+    private void monitorCommandEvent() {
+        // start monitor-command-example
+        MongoClientSettings settings =
+                MongoClientSettings.builder()
+                        .applyConnectionString(URI)
+                        .addCommandListener(new CommandTimer())
+                        .build();
+        MongoClient mongoClient = MongoClients.create(settings);
+        // Run some commands to test the timer
```

Comment:
Suggestion: I noticed with your other snippets, you place the comment for the tests being run right when it starts. For that reason, I think Line 39 should go right above your first test case.

Reply:
Thanks for the note. I agree this comment should be moved after creating the MongoCollection.

Fix: https://github.com/mongodb/docs-java/pull/88/commits/64b0850a3a4318719a9f6d9e6749baa281ab2153